src/tools/proxy.go

```diff
@@ -32,6 +52,10 @@ func ProxyRequestHandler(p *httputil.ReverseProxy) func(http.ResponseWriter, *ht
 		} else {
 			originalUrlResolver[r.RemoteAddr].Scheme = "http"
 		}
+		// Set URL with an updated path based on the request
+		log.Printf("baseURL  %s", CodeArtifactAuthInfo.Url)
+		CodeArtifactAuthInfo.Url = fmt.Sprintf("%s/%s", extractBaseURL(CodeArtifactAuthInfo.Url), strings.TrimPrefix(r.URL.Path, "/"))
```

Comment:
Haven't had some time to look at or test this code yet, but are you able to explain what problem is being solved with these changes?

I am likely missing something, but it looks like we're over-writing the ProxyRequestHandler hooks everytime a request comes through.

Reply:
Hello Steven Tan,

Current project status supports only the "Repository Type" value.
The change i made is using "Repository Type" only for the health check.
I am rewriting ProxyRequestHandler to support multiple types during runtime.
For example if a specific team would like pip and not maven, currently it is hard coded maven by the "Repository Type" ENV param. 

For example -
Currently status: proxy support only maven curl http://localhost:8080/maven/maven-central-store/group-id/artifact-id/version/artifact-id-version.jar.sha1
Change status:
The change allow to redirect request with artifact type
curl http://localhost:8080/maven/maven-central-store/group-id/artifact-id/version/artifact-id-version.jar.sha1
curl http://localhost:8080/pip/maven-central-store/group-id/artifact-id/version/artifact-id-version.jar.sha1
Allowing both pip and maven

Thanks

Tomer